models/tasks.js

```diff
@@ -94,6 +94,25 @@ const fetchTask = async (taskId) => {
   }
 }
 
+/**
+ * Fetch assigned self task
+ * @param taskId { string }: taskId which will be used to fetch the task
+ * @param id { string }: id to check task is assigned to self or not
+ * @return {Promsie<taskData|Object>}
+ */
+const fetchSelfTask = async (taskId, userId) => {
+  try {
+    const task = await tasksModel.doc(taskId).get()
+    const taskData = task.data()
+    if (!taskData) return { taskNotFound: true }
+    if (userId !== taskData.assignee) return { notAssignedToYou: true }
```

Comment:
Using optional chaining here would make sense. What if a task doesn't have `assignee`?

Reply:
I got a bit confused here. Sorry for the confusion.